src/vs/workbench/parts/tasks/electron-browser/runAutomaticTasks.ts

```diff
@@ -121,4 +125,46 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 		});
 	}
 
-}
\ No newline at end of file
+}
+
+export class AllowAutomaticTaskRunning extends Action {
+
+	public static readonly ID = 'workbench.action.tasks.allowAutomaticRunning';
+	public static readonly LABEL = nls.localize('workbench.action.tasks.allowAutomaticRunning', "Allow Automatic Tasks in this Folder");
+
+	constructor(
+		id: string, label: string,
+		@IStorageService private storageService: IStorageService
+	) {
+		super(id, label);
+	}
+
+	public run(event?: any): Promise<any> {
+		this.storageService.store(ARE_AUTOMATIC_TASKS_ALLOWED_IN_WORKSPACE, true, StorageScope.WORKSPACE);
+		return Promise.resolve(void 0);
+	}
+}
+
+export class DisallowAutomaticTaskRunning extends Action {
+
+	public static readonly ID = 'workbench.action.tasks.disallowAutomaticRunning';
+	public static readonly LABEL = nls.localize('workbench.action.tasks.disallowAutomaticRunning', "Disallow Automatic Tasks in this Folder");
+
+	constructor(
+		id: string, label: string,
+		@IStorageService private storageService: IStorageService
+	) {
+		super(id, label);
+	}
+
+	public run(event?: any): Promise<any> {
+		this.storageService.store(ARE_AUTOMATIC_TASKS_ALLOWED_IN_WORKSPACE, false, StorageScope.WORKSPACE);
+		return Promise.resolve(void 0);
+	}
+}
+
+const category = nls.localize('tasksCategory', "Tasks");
+const actionRegistry = Registry.as<IWorkbenchActionRegistry>(Extensions.WorkbenchActions);
+
+actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(AllowAutomaticTaskRunning, AllowAutomaticTaskRunning.ID, AllowAutomaticTaskRunning.LABEL), 'Tasks: Allow Automatic Tasks in Folder', category);
```

Comment:
I think the registrations should go in task.contribution.ts

Reply:
Yes. For now I wanted to keep it all together so that it's easier for Alex to rework the commands.

Reply:
I moved the registration as suggested.